pkg/operator/periodicbackupcontroller/periodicbackupcontroller.go

```diff
@@ -46,14 +52,17 @@ func NewPeriodicBackupController(
 	eventRecorder events.Recorder,
 	operatorImagePullSpec string,
 	accessor featuregates.FeatureGateAccess,
+	backupVarGetter backuphelpers.BackupVar,
 	backupsInformer factory.Informer) factory.Controller {
 
 	c := &PeriodicBackupController{
 		operatorClient:        operatorClient,
 		backupsClient:         backupsClient,
 		kubeClient:            kubeClient,
 		operatorImagePullSpec: operatorImagePullSpec,
+		backupVarGetter:       backupVarGetter,
 		featureGateAccessor:   accessor,
+		isDefaultBackup:       false,
```

Comment:
So in order to avoid using a CR for the `NoConfig`, and to be able to run the sidecar `only` when no backup CR is provided by the user, I rely on the following logic 

- If No Backup CR provided by the user
  - (1.1) If SideCar enabled { do nothing }
  - (1.2) If SideCar disabled { enable }
- If Backup CR provided, disable the backup CR 
  - (2.1) If SideCar enabled { disable }
  - (2.2) If SideCar disabled { do nothing }

Since enabling/disabling the side car required redeploying the staticPods, which leads to a period of API-server unavailability, I use this flag to avoid redeploying the static pod  in cases (1.1) and (2.2)

If there is a better approach plz let me know

Reply:
cc @dusk125 @hasbro17